modules/apps/27-interchain-accounts/types/account.go

```diff
@@ -66,20 +65,24 @@ func NewInterchainAccount(ba *authtypes.BaseAccount, accountOwner string) *Inter
 }
 
 // SetPubKey - Implements AccountI
-func (InterchainAccount) SetPubKey(pubKey crypto.PubKey) error {
+func (ia InterchainAccount) SetPubKey(pubKey crypto.PubKey) error {
 	return fmt.Errorf("not supported for interchain accounts")
```

Comment:
ditto, is this what the SDK does for accounts?

Reply:
Updated. This code was grandfathered in from the previous work on ICA. I assumed up to now that the override here for `SetPubKey` & `SetSequence` was necessary as an interchain account is controlled by another chains on-chain logic rather than a private key. Perhaps this isn't necessary though? Wdyt?

Reply:
Lets revisit during the internal audit. I'm wondering how this will interact with module accounts (if we even need this type)